### PR TITLE
[Feature] Table function table() supports HDFS

### LINKED ISSUE #21159: Simply load files from S3/HDFS
## Feature request

Now, we could load files from S3/HDFS by broker load and external file table.
### Broker Load
```
LOAD LABEL db0.label2 (     DATA INFILE("s3a://starrocks-common/fuwei/student.parquet")     INTO TABLE tbl_student     FORMAT AS "parquet"     (name, age)  ) WITH BROKER (     "aws.s3.access_key" = "xxx",     "aws.s3.secret_key" = "xxx",    "aws.s3.endpoint"="xxx.amazonaws.com" );
```
### External File Table 
```
CREATE EXTERNAL TABLE tbl_student_s3 (     name string,      id int )  ENGINE=file PROPERTIES  (     "path" = "xxx.parquet",      "format" = "parquet",      "aws.s3.access_key" = "xxx",     "aws.s3.secret_key" = "xxx",     "aws.s3.region" = "us-west-2"  );
```

These methods are not easy to use. Moreover, we need to know the schema of files before we load data.

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by rickif**
Fixes https://github.com/StarRocks/starrocks/issues/21159
This PR adds the support of reading files from HDFS, including the following changes:
1. execute RPC `get_file_schema` in pthread
2. refactor RPC `get_info`/`get_pulsar_info`
3. rename `_async_thread_pool` as `_load_rpc_pool`, which handles `get_file_schema`/`get_info`/`get_pulsar_info`
4. remove be config `internal_service_async_thread_num`
5. add metric `load_rpc_threadpool_size`

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by rickif** [comment]
@mergifyio rebase main

File: be/src/runtime/exec_env.cpp [hunk 1]
```diff
@@ -211,6 +211,12 @@ Status ExecEnv::_init(const std::vector<StorePath>& store_paths) {
     }
     _query_rpc_pool = new PriorityThreadPool("query_rpc", query_rpc_threads, std::numeric_limits<uint32_t>::max());
 
+    int load_rpc_threads = config::internal_service_async_thread_num;
```
Comment (chaoyli) [inline be/src/runtime/exec_env.cpp:214]: use `ThreadPool` to replace the `PriorityThreadPool`

Comment (rickif) [reply be/src/runtime/exec_env.cpp:214]: It seems there's no significant difference here. `PriorityThreadPool` is easier to use, and more widely used in our code base.

File: be/src/common/config.h [hunk 2]
```diff
@@ -885,8 +885,6 @@ CONF_mInt32(spill_init_partition, "16");
 // If the file size exceeds this limit, a new file will be created to store the block.
 CONF_Int64(spill_max_log_block_container_bytes, "10737418240"); // 10GB
 
-// Now, only get_info is processed by _async_thread_pool, and only needs a small number of threads.
-// The default value is set as the THREAD_POOL_SIZE of RoutineLoadTaskScheduler of FE.
 CONF_Int32(internal_service_async_thread_num, "10");
```
Comment (meegoo) [inline be/src/common/config.h:888]: using a dynamic thread pool limit concurrency by FE

Comment (rickif) [reply be/src/common/config.h]: A `threadpool` with min threads 10 and max threads 1000 is used.

File: be/src/runtime/exec_env.cpp (hunk 1 above)
Comment (rickif) [reply be/src/runtime/exec_env.cpp]: A threadpool with min threads 10 and max threads 1000 is used.

https://github.com/StarRocks/starrocks/blob/8860c4ab818e93476fed6b28fa74c2d7108ae0a0/be/src/runtime/exec_env.cpp#L215-220

**Review by chaoyli — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by rickif** [comment]
@mergifyio backport branch-3.1

**Pull request merged**